### PR TITLE
Add client side cache store.

### PR DESCRIPTION
**Pull request by zutigrm**
## Summary

Addresses issue:

- #6472 

## Relevant technical choices

Created new cache store per IB. Tests for the new action are included as well.

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/googlesitekit/datastore/site/cache.test.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
+/**
+ * `core/site` data store: client side cache tests.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
```
Comment (hussain-t) [inline assets/js/googlesitekit/datastore/site/cache.test.js:4]:
```suggestion
 * Site Kit by Google, Copyright 2023 Google LLC
```

File: assets/js/googlesitekit/datastore/site/cache.test.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	beforeEach( () => {
+		registry = createTestRegistry();
+	} );
+
+	afterEach( () => {
+		unsubscribeFromAll( registry );
+	} );
+
+	describe( 'actions', () => {
+		describe( '*setCacheItem', () => {
```
Comment (hussain-t) [inline assets/js/googlesitekit/datastore/site/cache.test.js:41]: This is not in our conventions. So, we could remove the asterisk.
```suggestion
		describe( 'setCacheItem', () => {
```

**Review by hussain-t — CHANGES_REQUESTED** [review]
Thanks, @zutigrm. Overall, the changes look good. I've left a few nitpicky comments. Please take a look. 

FYI, verifying the unit tests is part of the code review. This ticket doesn't involve any user-facing changes. Hence, it can bypass the QA and move to approval by the merge reviewer. I have updated the QAB.

**Review by hussain-t — APPROVED** [review]
Nice one, @zutigrm. LGTM 👍

**Review by tofumatt — APPROVED** [review]

File: assets/js/googlesitekit/datastore/site/cache.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+	 *
+	 * @since n.e.x.t
+	 * @see setItem
+	 *
+	 * @param {string}  key              Name of cache key.
+	 * @param {*}       value            Value to store in the cache.
+	 * @param {Object}  args             Optional object containing ttl, timestamp and isError keys.
+	 * @param {number}  [args.ttl]       Optional. Validity of the cached item in seconds.
+	 * @param {number}  [args.timestamp] Optional. Timestamp when the cached item was created.
+	 * @param {boolean} [args.isError]   Optional. Whether the cached item is an error.
```
Comment (aaemnnosttv) [inline assets/js/googlesitekit/datastore/site/cache.js:47]: For an arguments object with properties that are all optional, we use `options` as the name for this kind of object as a convention. `args` could contain optional properties but implies that at least one or more are required.

**Pull request merged**